### DISCUSSION TITLE
Filling forms and variety of fonts

### DISCUSSION THREAD

**Discussion by bfg01** [sumatrapdfreader/sumatrapdf#4422] [Q&A] [answered]
After searching thoroughly in the forums here and in bug reports, I saw that these issues already have many years being talked about and have never been actually solved. Seemingly the topics/reports are closed because there is no willing to discuss anymore.

If mupdf has this capability, why was it never implemented in sumatra?

I also asked about "fonts" because I'm searching free, lighter alternatives to the heavy-bloated adobe reader (just the reader), and sumatra would seem to be the most advanced for Windows platforms, but lacking these things in the title.

**Comment by GitHubRulesOK** [comment]
Form filling was bought by Adobe and then they did their own thing. So basically the fonts in forms are a hotchpotch of abilities but mainly auto sizing Latin characters. Unless your Adobe and even then its often hit and miss if its public method or private method/code they are using. Adobe Reader DC as an editor does not need fields to form fill, it will scan PDF for boxes and underlines in plain text too.
![image](https://github.com/user-attachments/assets/a160f3e7-acb5-4b7d-a44d-8311b5d017f2)


MuPDF does have the ability to fill prepared form fields or add comments with the Latin characters easily (like SumatraPDF uses), but depends on 3rd party support for fonts so there it is very basic ability but not same as Adobe. It is down to the Field Author to provide the best definition.
![image](https://github.com/user-attachments/assets/dc9b036e-a810-4fde-a39a-c0d59e648f12)
![image](https://github.com/user-attachments/assets/18b4f92f-aa73-43ea-bb6a-96a589a35d36)
![image](https://github.com/user-attachments/assets/5172a3c1-16cb-4efb-9ea1-a8c137a4de48)

MuPDF-GL cannot handle global characters. well this is wrong way round but if written well elsewhere it is supported 

![image](https://github.com/user-attachments/assets/9328b759-7d18-46fa-a20c-f025bb7be279)
![image](https://github.com/user-attachments/assets/9ec1984c-953f-4be3-9ee3-86247c41d6f4)![image](https://github.com/user-attachments/assets/7e29680d-e50d-4758-9dd6-1aa7ccdaece6)



Most Governmental or Major organisations that provide forms use Adobes own licensed software that mandates using Readers hidden abilities to edit forms using Adobe proprietary code.  Commonly yet often they forget to badge the forms with "This File MUST be edited in Acrobat Reader 9 or newer". Many of those forms based on Adobe XMP architecture are NOT public ISO standard PDF.

Or more accurately not 100% public PDF readers forms methodology.

**Comment by kjk** [comment, marked as answer]
Which fonts specifically? 

Fonts are large. Majority of size of SumatraPDF is not the code but fonts. I'm reluctant to add tens of megabytes of fonts.

Form filling is not implemented because no one spent the time to implement it. 

It really is that simple: someone has to write the code. You didn't and neither did anyone else.
